modules/swagger-codegen/src/main/resources/ruby/api_error.mustache

```diff
@@ -26,5 +26,24 @@ module {{moduleName}}
         super arg
       end
     end
+
+    # Override to_s to display a friendly error message
+    def to_s
+      message
+    end
+
+    def message
+      if @message.nil?
+        msg = "Error message: the server returns an error"
+      else
+        msg = @message
+      end
+
+      msg += "\nHTTP status code: #{code}" if code
+      msg += "\nResponse headers: #{response_headers}" if response_headers
+      msg += "\nResponse body: #{response_body}" if response_body
```

Comment:
So the error message is one long string? Why is it not an object? Like: 
```
error:{
   message: '...',
   status: 400,
   headers: ...,
   body: ...
}
```
?

Reply:
We want to output a string here but I can display it differently. 

The issue is that when you display the error as a string (ie. by doing `puts e`), nothing is displaying because `e.to_s` (implicitly called when you `puts`) seems to be defaulting to `error.message` (and message is empty). 

Example:

```
begin
  // API Call
rescue SmoochApi::ApiError => e
  puts e # blank
  puts e.to_s # blank
  puts e.message.to_s # blank
end
```

Reply:
The error itself is still an object btw.  Ie. the following worked, provided you inspected the error and knew what methods to call:

```
puts error.code
puts error.response_body
puts error.response_headers 
```